madprops/serializers.py

```diff
@@ -0,0 +1,140 @@
+from django.utils.functional import cached_property
+from rest_framework.serializers import (
+    ModelSerializer, ModelSerializerOptions, RelationsList)
+
+
+class PropertiesSerializerOptions(ModelSerializerOptions):
+    """Meta class options for PropertiesSerializer"""
+    def __init__(self, meta):
+        super(PropertiesSerializerOptions, self).__init__(meta)
+        self.parent_obj_name = getattr(meta, 'parent_obj_name', None)
+        self.read_only_props = getattr(meta, 'read_only_props', [])
+        self.exclude = ('id',)
+
+
+class NestedPropertiesSerializerOptions(PropertiesSerializerOptions):
+    """Meta class options for NestedPropertiesSerializer"""
+    def __init__(self, meta):
+        super(NestedPropertiesSerializerOptions, self).__init__(meta)
+        self.exclude = ('id', self.parent_obj_name)
+
+
+class PropertiesSerializer(ModelSerializer):
+    """Allows to operate on properties of certain resource as dictionary
+
+    Intended to be used as a base class for serializer for property resource
+    exposed via separate endpoint.
+
+    We consider Property as a model of the following structure:
+
+    class Property(model):
+        id = AutoField()
+        parent_model: ForeignKey(...)
+        name = CharField()
+        value = CharField()
+        id = models.AutoField()
+
+    to representation (converts Property instance(s) to dictionary):
```

Comment:
This sounds not very good with what you say above. Looks like you've updated text above, but left this as is.